packages/block-library/src/audio/edit.js

```diff
@@ -31,28 +30,34 @@ import { createUpgradedEmbedBlock } from '../embed/util';
 
 const ALLOWED_MEDIA_TYPES = [ 'audio' ];
 
-class AudioEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.toggleAttribute = this.toggleAttribute.bind( this );
-		this.onSelectURL = this.onSelectURL.bind( this );
-		this.onUploadError = this.onUploadError.bind( this );
-	}
+function AudioEdit( {
+	attributes,
+	noticeOperations,
+	setAttributes,
+	onReplace,
+	isSelected,
+	noticeUI,
+} ) {
+	// constructor() {
+	// 	super( ...arguments );
+	// 	this.toggleAttribute = this.toggleAttribute.bind( this );
+	// 	this.onSelectURL = this.onSelectURL.bind( this );
+	// 	this.onUploadError = this.onUploadError.bind( this );
+	// }
+	const { id, autoplay, caption, loop, preload, src } = attributes;
 
-	componentDidMount() {
-		const {
-			attributes,
-			mediaUpload,
-			noticeOperations,
-			setAttributes,
-		} = this.props;
-		const { id, src = '' } = attributes;
+	const audioUpload = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		const { mediaUpload } = getSettings();
+		return { mediaUpload };
+	} );
```

Comment:
You shouldn't be destructuring the `return` statement like that. That breaks the code. Also, you can avoid having to rename the variable to `audioUpload` if you return `getSettings.mediaUpload` directly.

Also, you're missing a dependency array here to memoize this `useSelect` call. In this case, the dependency array should be empty, since it doesn't rely on any external values.

```suggestion
	const mediaUpload = useSelect( ( select ) => {
		const { getSettings } = select( 'core/block-editor' );
		return getSettings().mediaUpload;
	}, [] );
```